R/preprocessing_pipeline.R

```diff
@@ -0,0 +1,34 @@
+preprocessing_pipeline <- function(choices  ,
+                                   text,
+                                   infrequent_term_threshold = .01,
+                                   verbose = FALSE,
+                                   save_dfm = FALSE,
+                                   intermediate_directory = NULL)
+{
+    stopwords <- if(choices$removeStopwords) quanteda::stopwords() else FALSE
+    if(choices$use_ngrams){ #need to tokenize before ngram before preprocessing
+        text <- quanteda::tokens(text)
+        text <- quanteda::tokens_ngrams(text, n = 1:3)
+    }
+    
+    text <- quanteda::dfm(text,
```

Comment:
I'd suggest using the `...` options passed to `tokens()` separately from the dfm stage, which provides more control and is more future-proof. i.e. https://github.com/quanteda/quanteda/issues/1602

Reply:
Thanks, I will do that. It would be interesting to try different stemming algorithm so that's a step in the good direction. 

Could you confirm that we first remove tokens then perform stemming such as : 

1. remove stopwords / punctuation /  numbers
4. stemming,
5. ngram
6. frequency

Reply:
Yes and I'd suggest adding `padding = TRUE` when you remove things, if ngrams != 1. Why? Because if you don't pad removed (stopwords) by a space, then new and potentially nonsensical ngrams are formed. Example:

```r
library("quanteda")
## Package version: 2.1.2
## Parallel computing: 12 of 12 threads used.
## See https://quanteda.io for tutorials and examples.
##
## Attaching package: 'quanteda'
## The following object is masked from 'package:utils':
##
##     View
tokens("united against the states") %>%
  tokens_remove(stopwords("en")) %>%
  tokens_ngrams()
## Tokens consisting of 1 document.
## text1 :
## [1] "united_states"
tokens("united against the states") %>%
  tokens_remove(stopwords("en"), padding = TRUE) %>%
  tokens_ngrams()
## Tokens consisting of 1 document.
## text1 :
## character(0) 
```

Reply:
Done! It has been implemented on the "multilang" branch. 

It's my first time working with multiple branch, should I merge "multilang" into this one? Or just delete this pull request.